osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -62,65 +51,174 @@ public SocialOverlay()
+
+            //currentQuery.ValueChanged += v =>
+            //{
+            //    queryChangedDebounce?.Cancel();
+
+            //    if (string.IsNullOrEmpty(v))
+            //        Scheduler.AddOnce(updateSearch);
+            //    else
+            //        queryChangedDebounce = Scheduler.AddDelayed(updateSearch, 500);
+            //};
+
+            //currentQuery.BindTo(Filter.Search.Current);
         }
 
         [BackgroundDependencyLoader]
         private void load(APIAccess api)
         {
-            if (Users == null)
-                reloadUsers(api);
+            this.api = api;
+            api.Register(this);
         }
 
-        private void reloadUsers(APIAccess api)
+        private void recreatePanels(PanelDisplayStyle displayStyle)
         {
-            Users = null;
+            clearPanels();
 
-            // no this is not the correct data source, but it's something.
-            var request = new GetUsersRequest();
-            request.Success += res =>
+            if (Users == null)
+                return;
+
+            var newPanels = new FillFlowContainer<SocialPanel>
             {
-                Users = res.Select(e => e.User);
-                loading.Hide();
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Spacing = new Vector2(10f),
+                Margin = new MarginPadding { Top = 10 },
+                ChildrenEnumerable = Users.Select(u =>
+                {
+                    SocialPanel panel;
+                    switch (displayStyle)
+                    {
+                        case PanelDisplayStyle.Grid:
+                            panel = new SocialGridPanel(u)
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre
+                            };
+                            break;
+                        default:
+                            panel = new SocialListPanel(u);
+                            break;
+                    }
+                    panel.Status.BindTo(u.Status);
+                    return panel;
+                })
             };
 
-            api.Queue(request);
+            LoadComponentAsync(newPanels, f =>
+            {
+                if(panels != null)
+                    ScrollFlow.Remove(panels);
+
+                ScrollFlow.Add(panels = newPanels);
+            });
+        }
+
+        private void clearPanels()
+        {
+            if (panels != null)
+            {
+                panels.FadeOut(200);
+                panels.Expire();
+                panels = null;
+            }
+        }
+
+        private APIRequest getUsersRequest;
+
+        //private readonly Bindable<string> currentQuery = new Bindable<string>();
+
+        //private ScheduledDelegate queryChangedDebounce;
+
+        private void updateSearch()
+        {
+            //queryChangedDebounce?.Cancel();

```

Comment:
are we gonna add this back before merging? debounce should be present at least for search text updates.